bleak/backends/dotnet/scanner.py

```diff
@@ -83,8 +85,50 @@ def AdvertisementWatcher_Received(self, sender, e):
             else:
                 if e.BluetoothAddress not in self._devices:
                     self._devices[e.BluetoothAddress] = e
+
         if self._callback is not None:
-            self._callback(sender, e)
+            # Get the default dictionary to populate
+            callback_data = get_reference_callback_format()
+
+            # Get any and all uuids from the advertisement
+            try:
+                service_uuids = []
+                for u in e.Advertisement.ServiceUuids:
+                    service_uuids.append(u.ToString())
+            except Exception:
+                logger.exception("Exception caught while parsing uuids from discovery")
+                service_uuids = []
+
+            # Get the manufacturer data from the advertisement, borrowed from parse_eventargs
+            try:
+                data = {}
+                for manufacturer in e.Advertisement.ManufacturerData:
+                    with BleakDataReader(manufacturer.Data) as reader:
+                        m_data = []
+                        m_data[:] = bytes(reader.read()).hex()
+                        data[manufacturer.CompanyId] = m_data
+            except Exception:
+                logger.exception(
+                    "Exception caught while parsing manufacturer data from discovery"
+                )
+                data = {}
+
+            try:
+                callback_data["address"] = e.BluetoothAddress or None
+                callback_data["name"] = e.Advertisement.LocalName or None
+                callback_data["data_channel"] = None
+                callback_data["manufacturer_data"] = data
+                callback_data["service_data"] = None
```

Comment:
I think we should be able to get this with `e.Advertisement.GetSectionsByType(0x16)` plus some additional processing. And 0x20 and 0x21 are probably needed too to be complete.